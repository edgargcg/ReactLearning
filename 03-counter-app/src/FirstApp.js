import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

// functional components
const FirstApp = ({ name, subtitle }) => {
    const greetings = `Hello World ${name}`;

    return (
        <Fragment>
            <div className="container">
                <h1>
                    { greetings }
                </h1>
                <h4>{ subtitle }</h4>
            </div>
        </Fragment>
    );
}

FirstApp.propTypes = {
    name: PropTypes.string.isRequired,
    subtitle: PropTypes.string
};

FirstApp.defaultProps = {
    subtitle: 'I am a subtitle'
}

export default FirstApp;