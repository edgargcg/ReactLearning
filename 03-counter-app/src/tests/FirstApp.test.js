import React from 'react';
import { render } from "@testing-library/react";
import { shallow } from 'enzyme';
import FirstApp from "../FirstApp";
import '@testing-library/jest-dom';

describe('Test on <FirstApp />', () => {

    // test('should render message', () => {
    //     const name = 'Edgar';
    //     const greeting = `Hello World ${name}`;

    //     const { getByText } = render(<FirstApp name={name} />);

    //     expect(getByText(greeting)).toBeInTheDocument();
    // })

    test('should render <FirstApp />', () => {
        const name = 'Edgar';
        const wrapper = shallow(<FirstApp name={name} />);
        
        expect(wrapper).toMatchSnapshot();
    })   
    
    test('should render subtitle in props', () => {
        const name = 'Edgar';
        const subtitle = "Let's play!!";
        const wrapper = shallow(
            <FirstApp 
                name={name} 
                subtitle={subtitle}
            />
        );
        
        const subtitleElementText = wrapper.find('h4').text();

        expect(subtitleElementText).toBe(subtitle);
    })
    
    
})
