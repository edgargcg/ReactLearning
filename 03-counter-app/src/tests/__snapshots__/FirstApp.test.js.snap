// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <FirstApp /> should render <FirstApp /> 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h1>
      Hello World Edgar
    </h1>
    <h4>
      I am a subtitle
    </h4>
  </div>
</Fragment>
`;

exports[`Test on <FirstApp /> should render subtitle in props 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h1>
      Hello World Edgar
    </h1>
    <h4>
      I am a subtitle
    </h4>
  </div>
</Fragment>
`;
